pkg/bpf/prog.go

```diff
@@ -67,3 +74,72 @@ func (t ProgType) String() string {
 
 	return "Unknown"
 }
+
+type attrProg struct {
+	startID   uint32
+	progID    uint32
+	nextID    uint32
+	openFlags uint32
+}
+
+// ProgInfo holds values from the upstram struct bpf_prog_info.
+// From: https://github.com/torvalds/linux/blob/master/include/uapi/linux/bpf.h#L1006
+type ProgInfo struct {
+	ProgType     ProgType
+	ID           uint32
+	MapIDs       uint64
```

Comment:
Would that be passed / used in GetObjInfoByFD() as info, since mapping to `struct bpf_prog_info` is different? Note the __aligned_u64 is actually a pointer to a __u32 array, in the uapi header it's done as such to avoid having to implement the compat layer for it in kernel to cope with 32 bit uspace on 64 bit kernel. https://github.com/torvalds/linux/blob/v4.16/tools/bpf/bpftool/prog.c#L185